### PR TITLE
Use prebuilt ruby binaries instead of building on the fly

### PR DESCRIPTION
**Pull request by kylekthompson**
This change introduces an out-of-band run definition that we'll use to perform a Ruby installation. Once that installation is done, we fetch the layer produced by that task, push its tarball to S3, and store it permanently as a prebuilt binary.

When the `mint/install-ruby` leaf runs, it fetches hardcoded-per-release tarballs from that bucket and expands them much like docker itself might (by loosely following https://github.com/opencontainers/image-spec/blob/main/layer.md#applying-changesets)

### CODE REVIEW

File: .mint/ruby-binary-builder.yml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+      export XDG_CACHE_HOME="./tmp/cache"
+      source_dir="./tmp/src"
+
+      function cleanup {
+        # TODO: mintignore these instead of removing
+        echo "Cleaning up"
+        sudo rm -rf "$source_dir" "$XDG_CACHE_HOME"
+      }
+      trap cleanup EXIT
+
```
Comment (ayazhafiz) [inline .mint/ruby-binary-builder.yml:42]: I think this will overwrite the trap for EXIT defined on line 20.

Comment (kylekthompson) [reply .mint/ruby-binary-builder.yml:42]: ope, you're right. I should've more closely looked at these when I merged them

File: mint/install-ruby/mint-leaf.yml [hunk 2]
```diff
@@ -11,15 +11,67 @@ parameters:
...
+
+      # Find the prebuilt Ruby
+      echo "Looking for Ruby $RUBY_VERSION..."
+      escaped_version=$(echo "$RUBY_VERSION" | sed "s/\./\\\\./g") # 3.2.2 => 3\.2\.2
+      version_line=$(cat $MINT_LEAF_PATH/known-rubies.csv | grep "^$escaped_version,")
+      if [[ -z "$version_line" ]]; then
+        >&2 echo "Ruby $RUBY_VERSION is not available in this version of mint/install-ruby."
+        >&2 echo "Try upgrading to the most recent version of this leaf. If the version is still not available, file an issue at https://github.com/rwx-research/mint-leaves/issues/new."
+        exit 3
+      fi
```
Comment (ayazhafiz) [inline mint/install-ruby/mint-leaf.yml:40]: You could use `$MINT_ERRORS` here! For example https://github.com/rwx-research/mint/blob/cee77a920b879c6ba9ea9fead2132d9d76dc424b/packages/bigtest/errors.test.bigtest.ts#L23-L28

Comment (kylekthompson) [reply mint/install-ruby/mint-leaf.yml:40]: Oh nice! I'll switch this over

**Pull request closed without merging**